```{r}
library(data.table)
library(ggplot2)
library(reshape2)
strsplit2 <- function (x, split, ...) 
{
    x <- as.character(x)
    n <- length(x)
    s <- strsplit(x, split = split, ...)
    nc <- unlist(lapply(s, length))
    out <- matrix("", n, max(nc))
    for (i in 1:n) {
        if (nc[i]) 
            out[i, 1:nc[i]] <- s[[i]]
    }
    out
}
setwd("~/temp/MIX/")

```


```{r}

lengths<-fread("../mollicutes_core_genome_cluster_of_proteins.protein.length.tsv")
setnames(lengths,c("q","q.length"))
lengths$cluster<-factor(strsplit2(lengths$q,split="_")[,1])

n.prot.by.clust <- lengths[,.N,by=cluster]
sorted.clusters <- factor(n.prot.by.clust$q,levels=n.prot.by.clust$cluster[order(n.prot.by.clust$N)])


setwd("~/temp/MIX/core_genome/MBVG/")


blast.cols <- c("q", "contig", "identity", "alignment.length", "mismatches", "gap.openings", "q.start", "q.end", "s.start", "s.end", "e.value", "bit.score","method")


all.results <- data.table()
for(a.file in list.files(pattern="hits")){
  cat(a.file,"\n")
  this.results <- fread(a.file)
  this.results$method = a.file 
  all.results<-rbind(all.results,this.results)
}

prepare.hit.table <- function(an.hit.table){
  setnames(an.hit.table,colnames(an.hit.table),blast.cols)
  an.hit.table$cluster<-factor(strsplit2(an.hit.table$q,split="_")[,1])
  an.hit.table<-merge(an.hit.table,lengths[,list(q,q.length)],by="q")
  an.hit.table[,protein.found.50:=(alignment.length/q.length)>=0.5]
  an.hit.table[,protein.found.85:=(alignment.length/q.length)>=0.85]
  an.hit.table[,protein.found.99:=(alignment.length/q.length)>=0.99]
  an.hit.table$e.value<-as.numeric(an.hit.table$e.value)
  an.hit.table$assemblies <- strsplit2(strsplit2(an.hit.table$method,"_")[,2],".",fixed=T)[,1]
  return(an.hit.table)
  }


all.results <- prepare.hit.table(all.results)


ident.thr <- 100
all.results[identity>=ident.thr,.N,by=list(cluster,method,assemblies)][N>=1][,.N,by=list(method,assemblies)][order(N)]

ggplot(data=all.method[identity>=50 & method %in% c("MIX","GAM")],aes(x=as.numeric(cluster),y=identity,colour=method))+geom_point(alpha=0.3)+geom_smooth()+facet_wrap(~method)
ggplot(data=all.results[identity>=99 & method %in% c("MIX","GAM","CLC")],aes(x=as.numeric(cluster),y=identity,colour=method))+geom_point(alpha=0.3)+geom_smooth()
```


* Check number of contigs without proteins ?
```{r}
all.results[,length(unique(contig)),by=method]
```


* Check against published 
```{r}

maure.mira <- fread("~/temp/MIX/datasets/Mollicutes/MAUR/MAUR_MIRA_out.unpadded.hits")
maure.clc <- fread("~/temp/MIX/datasets/Mollicutes/MAUR/MAUR_CLC_contigsCLCTrimmed_0_vs_MAUR_proteome.hits")
maure.sub <- fread("~/temp/MIX/datasets/Mollicutes/MAUR/MAUR_submitted.hits")
setnames(maure.clc,c("q", "contig", "identity", "alignment.length", "mismatches", "gap.openings", "q.start", "q.end", "s.start", "s.end", "e.value", "bit.score"))
setnames(maure.mira,c("q", "contig", "identity", "alignment.length", "mismatches", "gap.openings", "q.start", "q.end", "s.start", "s.end", "e.value", "bit.score"))
setnames(maure.sub,c("q", "contig", "identity", "alignment.length", "mismatches", "gap.openings", "q.start", "q.end", "s.start", "s.end", "e.value", "bit.score"))
```



New computation with corrected genetic code for Mycoplasmas 
============================================================
```{r}
# core <- fread("core_genome/Core_all.0.tsv")
core <- fread("core_genome/mycoplasmas_all_core_conservation.tsv")
core[,rankP99:=rank(-p99.99,ties="first"),by=list(Specie,Merger)]
core[,meanP99:=mean(p99.99),by=list(Specie,Merger)]
core[,maxP99:=max(p99.99),by=list(Specie,Merger)]
core[,minP99:=min(p99.99),by=list(Specie,Merger)]
core$medianP99<-median(core$p99.99)
core.m <- data.table(melt(core[rankP99==1,]))[variable!="rankP99"]

ggplot(core.m,aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+facet_wrap(~variable)
ggplot(core.m[variable=="p85"],aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+coord_cartesian(ylim=c(100,170))
ggplot(core.m[variable=="p99.99"],aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+coord_cartesian(ylim=c(100,150))
ggplot(core.m[variable=="meanP99"],aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+coord_cartesian(ylim=c(50,150))+ylab("Mean number of proteins recovered with 99% identity")
ggsave("figures/core.genome_core0_mean.pdf",w=12,h=8)
ggplot(core.m[variable=="medianP99"],aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+coord_cartesian(ylim=c(50,150))
ggplot(core.m[variable=="maxP99"],aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+coord_cartesian(ylim=c(50,150))
ggplot(core.m[variable=="minP99"],aes(x=Specie,y=value,fill=Merger))+geom_bar(stat="identity",pos="dodge")+coord_cartesian(ylim=c(50,150))+ylab("Min number of proteins with 99% identity")
ggsave("figures/core.genome_core0_min.pdf",w=12,h=8)

```
```{r}
ggplot(core,aes(x=Specie,fill=Merger,y=p99.99))+geom_boxplot(alpha=0.5,outlier.size=0)+coord_cartesian(ylim=c(80,150))+ylab("Number of clusters with a conserved representative(Identity>=99.99%)")+scale_fill_manual(values=c("#7FC97F","#E69F00", "#56B4E9","#999999"))
#+scale_fill_manual(values=brewer.pal("Accent",n=4))

ggplot(core,aes(x=Specie,fill=Merger,y=p99.99,colour=Merger))+ stat_boxplot(geom ='errorbar')+geom_point(position=position_jitter(width=0.2))+coord_cartesian(ylim=c(80,150))
#+scale_fill_manual(values=brewer.pal("Accent",n=4))


ggsave("figures/core.genome_core0_boxes_v2.pdf",w=10,h=7)
library(RColorBrewer)

```